www/templates/html/MediaPlayer.tpl.php

```diff
@@ -4,22 +4,162 @@
+                                $transcript.find("li").addClass("highlight");
+                            });
+
+                            $captionSearch.find(".mh-parse-caption").on("keyup focus", function(e){
+                                e.stopPropagation();
+                                var search = $(this).val().toLowerCase();
+                                var subtitlesLength;
+                                var i;
+                                if (!search) {
+                                    return;
+                                }
+
+                                subtitlesLength = track.entries.text.length;
+                                for (i = 0; i < subtitlesLength; i++) {
+                                    var line = track.entries.text[i].toLowerCase();
+                                    if (line.indexOf(search) > -1){
+                                        $transcript.find("li").eq(i).addClass("highlight");
+                                    }else{
+                                        $transcript.find("li").eq(i).removeClass("highlight");
+                                    }
+                                };
+                            });
+
+                            t.container.find(".mh-paragraph-icons").off();
+                            t.container.find(".mh-paragraph-icons").on("click", function(){
+                            t.container.find(".mh-caption-search").toggleClass("bulleted");
+
+                            });
+
+                            $transcript.on('click', 'li', function() {
+                                var time;
+                                time = $(this).data('timeOffset')
+                                if (!time) {
+                                    return;
+                                }
+                                t.setCurrentTime(time);
+                            });
+                            var listItems = [];
+                            for (var i = 0; i < track.entries.text.length; i++) {
+                                listItems.push($('<li>',  {"class": "highlight"})
+                                    .data('timeOffset', track.entries.times[i].start)
+                                    .text(track.entries.text[i])
+                                    .prepend($('<span>').text('[' + displaytime(track.entries.times[i].start*1000) + '] '))
+                                );
+                            };
+                            $transcript.children("li").remove();
+                            $transcript.append(listItems);
+                        };
+
+                        var origsEnableTrackButton = t.enableTrackButton;
+                        t.enableTrackButton = function(lang, label) {
+                            origsEnableTrackButton.call(this, lang, label);
+                            var t = this;
+                            setTranscript(t.tracks[0]);
```

Comment:
This still assumes that the first track will be set for _every_ track that is enabled (`enableTrackButton` is called after each track is successfully, sequentially loaded). If you only care about the first loaded track, you may want to make this function override revert back to the original after first run.